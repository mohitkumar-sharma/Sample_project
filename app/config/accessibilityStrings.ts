export default {
  NORMAL_BUTTON_TEXT: 'normal-button-text',
  SLIDER_BUTTON_TEXT: 'slider-button-text',
  NORMAL_BUTTON: 'normal-button',
  SLIDER_BUTTON_ICON: 'slider-button-icon',
  SLIDER_BUTTON_CONTAINER: 'slider-button-container',
  BANNER_TEXT: 'banner-text',
  HEADER_SIDES_TEXT: 'header-sides-text',
  HEADER_PRESSABLE_ELEMENT: 'header-pressable-element',
  SETTINGS_SWITCH_TEXT: 'settings-switch-text',
  SETTINGS_SWITCH: 'settings-switch',
  INV_TEXT_INPUT: 'inv-text-input',
};
