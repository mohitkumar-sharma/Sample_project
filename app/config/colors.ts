export default {
  black: '#000000',
  gray: '#9B9B9B',
  divider: '#E8E8E8',
  white: '#FFFFFF',
  themeBlue: '#6EB1F7',
  themeLightBlack: '#34424A',
  appTheme: '#1A1924',
  borderLine: '#59606A',
  warning: '#FFCC00',
  red: '#FF0000',
};
