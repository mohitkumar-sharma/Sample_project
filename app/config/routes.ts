export default {
  WELCOME: 'Welcome',
  HOME: 'Home',
  SETTINGS: 'Settings',
};
