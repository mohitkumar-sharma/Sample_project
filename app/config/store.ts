import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { rootReducer } from '../states';

const store = configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware(),
});

export type AppDispatch = typeof store.dispatch;

export { store };
