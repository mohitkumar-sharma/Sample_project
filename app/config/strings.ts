export default {
  GO_TO_HOME: 'Go To Home',
  DIAMOND_ICON: 'diamond',
  SLIDE_ME: 'Slide me to continue',
  APP_RUNNING_WARNING: 'Notice: You are running this app on',
  WELCOME_INPUT_PLACEHOLDER: 'Enter your name here...',
  SIGN_OUT: 'SignOut',
  HOME_BUTTONS_HEADING: '4 variations of a button',
  PRESS_ME: 'Press me',
  TRANSPARENT_BTN_PRESSED: 'Transparent button is pressed',
  LIGHT_BLACK_BTN_PRESSED: 'Light black button is pressed',
  BLUE_BTN_PRESSED: 'Blue button is pressed',
  SWITCH_TEXT: 'Enable Warning Banner',
  ALERT_TITLE_INVALID_USER_NAME: 'Invalid UserName!',
  ALERT_TITLE_BTN_PRESSED: 'Button Pressed!',
  ALERT_ENTER_VALID_USER_NAME: 'Please enter a valid user name.',
};
