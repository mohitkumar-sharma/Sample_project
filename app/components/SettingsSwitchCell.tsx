import React from 'react';
import { View, StyleSheet, StyleProp, ViewStyle, Text, TextStyle, Switch } from 'react-native';
import config from '../config';

export interface SettingsSwitchCellProps {
  /* as a string for showing text for Text element */
  text?: string;
  /* as an object of style props for main View element */
  containerStyle?: StyleProp<ViewStyle>;
  /* as an object of style props for Text element */
  textStyle?: StyleProp<TextStyle>;
  /* as an object of style props for Switch element */
  switchStyle?: StyleProp<TextStyle>;
  /* as a function initiates when value changes for Switch element*/
  onSwitchValueChange?: (value: boolean) => void;
  /* as a boolean to show the value on Switch element */
  switchValue?: boolean;
}

/**
 * This is the SettingsSwitchCell functional component works as a cell with a switch
 * @param props : props are defined in SettingsSwitchCellProps which can be passed to this component
 */
const SettingsSwitchCell = (props: SettingsSwitchCellProps): JSX.Element => {
  const { text, containerStyle, textStyle, switchStyle, onSwitchValueChange, switchValue } = props;
  return (
    <View style={[styles.container, containerStyle]}>
      {text && text !== '' ? (
        <Text
          style={[styles.text, textStyle]}
          testID={config.accessibilityStrings.SETTINGS_SWITCH_TEXT}>
          {text}
        </Text>
      ) : null}
      {onSwitchValueChange ? (
        <Switch
          style={[styles.switch, switchStyle]}
          value={switchValue}
          onValueChange={onSwitchValueChange}
          trackColor={{
            true: config.colors.appTheme,
            false: config.colors.divider,
          }}
          testID={config.accessibilityStrings.SETTINGS_SWITCH}
        />
      ) : null}
    </View>
  );
};

const styles = StyleSheet.create({
  container: { flexDirection: 'row', alignItems: 'center' },
  text: { flex: 1, paddingVertical: 5, fontSize: 15, marginHorizontal: 15 },
  switch: { marginRight: 15 },
});

export default SettingsSwitchCell;
