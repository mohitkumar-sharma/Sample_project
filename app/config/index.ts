import colors from './colors';
import localConfig from './localConfig';
import strings from './strings';
import routes from './routes';
import fontWeights from './fontWeights';
import accessibilityStrings from './accessibilityStrings';
import { store } from './store';

const config = {
  colors,
  ...localConfig,
  store,
  strings,
  routes,
  fontWeights,
  accessibilityStrings,
};

export default config;
