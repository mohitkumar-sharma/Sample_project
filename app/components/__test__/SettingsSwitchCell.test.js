import React from 'react';
import { render } from '@testing-library/react-native';
import SettingsSwitchCell from '../SettingsSwitchCell';
import config from '../../config';

describe('SettingsSwitchCell', () => {
  const props = {
    containerStyle: { padding: 10 },
    textStyle: { color: config.colors.appTheme },
    switchStyle: { marginLeft: 20 },
    switchValue: true,
  };
  const text = config.strings.SWITCH_TEXT;
  const onSwitchValueChange = jest.fn();

  it('matches snapshot', () => {
    expect(render(<SettingsSwitchCell {...props} />)).toMatchSnapshot();
    expect(
      render(
        <SettingsSwitchCell {...props} text={text} onSwitchValueChange={onSwitchValueChange} />
      )
    ).toMatchSnapshot();
  });

  it('renders the text', () => {
    const { queryByTestId } = render(<SettingsSwitchCell {...props} text={text} />);
    expect(queryByTestId(config.accessibilityStrings.SETTINGS_SWITCH_TEXT)).toBeTruthy();
  });

  it('does not render the text', () => {
    const { queryByTestId } = render(<SettingsSwitchCell {...props} />);
    expect(queryByTestId(config.accessibilityStrings.SETTINGS_SWITCH_TEXT)).toBeFalsy();
  });

  it('renders the switch', () => {
    const { queryByTestId } = render(
      <SettingsSwitchCell {...props} onSwitchValueChange={onSwitchValueChange} />
    );
    expect(queryByTestId(config.accessibilityStrings.SETTINGS_SWITCH)).toBeTruthy();
  });

  it('does not render the switch', () => {
    const { queryByTestId } = render(<SettingsSwitchCell {...props} />);
    expect(queryByTestId(config.accessibilityStrings.SETTINGS_SWITCH)).toBeFalsy();
  });
});
