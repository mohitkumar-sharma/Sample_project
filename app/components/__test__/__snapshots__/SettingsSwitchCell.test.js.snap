// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsSwitchCell matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
      Object {
        "padding": 10,
      },
    ]
  }
/>
`;

exports[`SettingsSwitchCell matches snapshot 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
      Object {
        "padding": 10,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "flex": 1,
          "fontSize": 15,
          "marginHorizontal": 15,
          "paddingVertical": 5,
        },
        Object {
          "color": "#1A1924",
        },
      ]
    }
    testID="settings-switch-text"
  >
    Enable Warning Banner
  </Text>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#1A1924"
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Array [
          Object {
            "marginRight": 15,
          },
          Object {
            "marginLeft": 20,
          },
        ],
      ]
    }
    testID="settings-switch"
    tintColor="#E8E8E8"
    value={true}
  />
</View>
`;
